File: solutions/beecrowd/1040/1040.py
n1, n2, n3, n4 = map(float, input().split())

average = (n1 * 2 + n2 * 3 + n3 * 4 + n4) / 10

print(f'Media: {average:.1f}')
if average >= 7.0:
    print('Aluno aprovado.')
elif average >= 5.0:
    print('Aluno em exame.')

    final_test = float(input())

    print(f'Nota do exame: {final_test:.1f}')
    if (final_test + average) / 2.0 > 5.0:
        print('Aluno aprovado.')
    else:
        print('Aluno reprovado.')
    print(f'Media final: {(final_test + average) / 2.0:.1f}')
else:
    print('Aluno reprovado.')

File: solutions/beecrowd/1040/1040.clj
(ns main
  (:require [clojure.string :as str]))

(defn retake-exam [average]
  (println "Aluno em exame.")
  (let [retake-grade (Double/parseDouble (read-line))
        final-average (/ (+ retake-grade average) 2.0)]
    (printf "Nota exame: %.1f%n" retake-exam)
    (if (> final-average 5.0)
      (println "Aluno aprovado.")
      (println "Aluno reprovado."))
    (printf "Media final: %.1f%n" final-average)))

(defn grade [n1 n2 n3 n4]
  (let [average (/ (+ (* n1 2) (* n2 3) (* n3 4) n4) 10)]
    (printf "Media: %.1f%n" average)
    (cond (>= average 7.0)
          (println "Aluno aprovado.")
          (>= average 5.0)
          (retake-exam average)
          :else
          (println "Aluno reprovado."))))

(defn main []
  (loop [line (read-line)]
    (when line
      (let [[n1 n2 n3 n4] (->> (str/split line #" ")
                               (map #(Double/parseDouble %)))]
        (grade n1 n2 n3 n4))
      (recur (read-line)))))

(main)
